packages/js/data/src/onboarding/with-onboarding-hydration.tsx

```diff
@@ -20,39 +19,26 @@ export const withOnboardingHydration = ( data: {
 	return createHigherOrderComponent< Record< string, unknown > >(
 		( OriginalComponent ) => ( props ) => {
 			const onboardingRef = useRef( data );
-
-			useSelect(
-				// @ts-expect-error // @ts-expect-error registry is not defined in the wp.data typings
-				( select: ( s: string ) => OnboardingSelector, registry ) => {
-					if ( ! onboardingRef.current ) {
-						return;
-					}
-
-					const { isResolving, hasFinishedResolution } =
-						select( STORE_NAME );
-					const {
-						startResolution,
-						finishResolution,
-						setProfileItems,
-					} = registry.dispatch( STORE_NAME );
-
-					const { profileItems } = onboardingRef.current;
-
-					if (
-						profileItems &&
-						! hydratedProfileItems &&
-						! isResolving( 'getProfileItems', [] ) &&
-						! hasFinishedResolution( 'getProfileItems', [] )
-					) {
-						startResolution( 'getProfileItems', [] );
-						setProfileItems( profileItems, true );
-						finishResolution( 'getProfileItems', [] );
-
-						hydratedProfileItems = true;
-					}
-				},
-				[]
-			);
+			const { startResolution, finishResolution, setProfileItems } =
+				useDispatch( STORE_NAME );
+			useEffect( () => {
+				if ( ! onboardingRef.current ) {
+					return;
+				}
+
+				const { profileItems } = onboardingRef.current;
+				if ( ! profileItems ) {
+					return;
+				}
+
+				if ( profileItems && ! hydratedProfileItems ) {
```

Comment:
Given the existing `profileItems` check above, the additional isn't needed.

Reply:
We've found that it actually broke redirects p1681188908953349/1681170705.881689-slack-C01SFMVEYAK

Added back the checks in https://github.com/woocommerce/woocommerce/pull/37641/commits/a0f40d1324ca0342bba43a74907794892098d480